src/dialects/abstract/query.js

```diff
@@ -531,6 +531,10 @@ class AbstractQuery {
       }
       return lastKeyPrefixMemo[key];
     };
+
+    // sort the array by the level of their depth calculated by dot.
+    const sortByDepth = keys => keys.sort((a, b) => a.split('.').length - b.split('.').length);
```

Comment:
Stable sorting wasn't a thing until Node 12 so I think that's why some tests fail on Node 10. Can you take a look?

Reply:
yeah sure

Reply:
Done @WikiRik  : )